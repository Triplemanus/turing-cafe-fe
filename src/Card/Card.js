import React from 'react';
import './Card.scss';
import PropTypes from 'prop-types';

const ResyCard = ({ id, name, date, time, number }) => {
  return (
    <section className="card">
      <article className="card-title">
        <h3> Name: <span>{name}</span></h3>
      </article>
      <article className="card-contents">
        <h3>Date: <span>{date}</span></h3>
        <h3>Time: <span>{time}</span></h3>
        <h3>Number: <span>{number}</span></h3>
      </article>
    </section>
  )
}

export default ResyCard;

ResyCard.propTypes = {
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
    time: PropTypes.string.isRequired,
    number: PropTypes.string.isRequired
  }