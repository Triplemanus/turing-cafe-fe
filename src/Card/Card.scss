.card {
  box-sizing: border-box;
  margin: 10px;
  height: 220px;
  width: 250px;
  background: rgb(211, 210, 210);
  padding-left: 10px;
  color:teal;
  border: solid 3px rgb(204, 203, 203);
}

.card_content {
  height: 150px;
}

.card-title {
  height: 50px;
  width: 95%;
  display: flex;
  justify-content: space-between;
}

.star-card-button {
  display: flex;
  height: 50px;
  margin-left: auto;
}

.star-card-button:active {
  display: flex;
  height: 50px;
  margin-left: auto;
}
