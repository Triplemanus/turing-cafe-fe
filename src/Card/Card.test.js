import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<Card
      name="Frankenstein"
      date="10/31"
      id="30"
      time="24:00"
      number="30"
    />);

    expect(wrapper).toMatchSnapshot();
  });
})