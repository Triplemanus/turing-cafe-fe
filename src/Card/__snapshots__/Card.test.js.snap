// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all data passed in correctly 1`] = `
<section
  className="card"
>
  <article
    className="card-title"
  >
    <h3>
       Name: 
      <span>
        Frankenstein
      </span>
    </h3>
  </article>
  <article
    className="card-contents"
  >
    <h3>
      Date: 
      <span>
        10/31
      </span>
    </h3>
    <h3>
      Time: 
      <span>
        24:00
      </span>
    </h3>
    <h3>
      Number: 
      <span>
        30
      </span>
    </h3>
  </article>
</section>
`;
